task/runner.go

```diff
@@ -221,6 +221,22 @@ func (r *runner) getAssetAndOS(assetID string) (*api.Asset, drivers.OSSession, e
 
 func (r *runner) publishTaskResult(ctx context.Context, task data.TaskInfo, output *data.TaskOutput, err error) error {
 	resultCh := make(chan event.PublishResult, 1)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "unexpected EOF") {
```

Comment:
Should probably get the `err.Error()` only once in the beginning, assign it to some variable and then just use it in these checks. 

It's not a good practice to call the same function over and over if you only need it's value once, since sometimes it can be just a waste in performance. In this case `Error()` functions are _generally_ very cheap, at worst only formatting some message, but it still leads to a repetition in the code that can be confusing, as if the response wouldn't always be the same from all those calls.

Also good to do a `strings.ToLower` on that error message so you don't need to worry about casing when checking for the error messages (tho make sure your string literals are all lowercase here): https://github.com/livepeer/go-api-client/blob/832e50712078b48755c17e203fa2f0433a98de93/api.go#L1116